Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,493 @@
+                self.torrents[tdict['infohash']]['num_seeders'] = new_seed
+                self.torrents[tdict['infohash']]['num_leechers'] = new_leecher
+                self._logger.info("infohash %s : seeder/leecher changed seed:%d leech:%d",
+                                  infohash_str, new_seed, new_leecher)
+
+    def scrape_trackers(self):
+        """
+        Manually scrape tracker by requesting to tracker manager
+        """
+
+        for infohash in list(self.torrents):
+            # torrent handle
+            lt_torrent = self.session.lm.ltmgr.get_session().find_torrent(lt.big_number(infohash))
+
+            peer_list = []
+            for i in lt_torrent.get_peer_info():
+                peer = LibtorrentDownloadImpl.create_peerlist_data(i)
+                peer_list.append(peer)
+
+            num_seed, num_leech = utilities.translate_peers_into_health(peer_list)
+            self._logger.debug("Seeder/leecher data translated from peers : seeder %s, leecher %s", num_seed, num_leech)
+
+            # check health(seeder/leecher)
+            self.session.lm.torrent_checker.add_gui_request(infohash, True)
+
+    def set_archive(self, source, enable):
+        """
+        setting archive of a particular source. Affect all the torrents in this source
+        """
+        if source in self.boosting_sources:
+            self.boosting_sources[source].archive = enable
+            self._logger.info("Set archive mode for %s to %s", source, enable)
+        else:
+            self._logger.error("Could not set archive mode for unknown source %s", source)
+
+    def start_download(self, torrent):
+        """
+        Start downloading a particular torrent and add it to download list in Tribler
+        """
+        dscfg = DownloadStartupConfig()
+        dscfg.set_dest_dir(self.settings.credit_mining_path)
+        dscfg.set_safe_seeding(False)
+
+        preload = torrent.get('preload', False)
+
+        pstate = CallbackConfigParser()
+        pstate.add_section('state')
+        pstate.set('state', 'engineresumedata', torrent.get('pstate', None))
+
+        # not using Session.start_download because we need to specify pstate
+        if self.session.lm.download_exists(torrent["metainfo"].get_infohash()):
+            self._logger.error("Already downloading %s. Cancel start_download",
+                               hexlify(torrent["metainfo"].get_infohash()))
+            return
+
+        self._logger.info("Starting %s preload %s has pstate %s",
+                          hexlify(torrent["metainfo"].get_infohash()), preload,
+                          True if torrent.get('pstate', None) else False)
+
+        torrent['download'] = self.session.lm.add(torrent['metainfo'], dscfg, pstate=pstate,
+                                                  hidden=True, share_mode=not preload, checkpoint_disabled=True)
+        torrent['download'].set_priority(torrent.get('prio', 1))
+
+    def stop_download(self, torrent):
```

Comment:
Just a minor detail, there is a difference between stopping and removing a download. In this case, you are removing a download (so it does not exist anymore). I think `stop_download` should be renamed to `remove_download` to avoid confusion.